groups/bdl/bdlb/bdlb_pcgrandomgenerator.h

```diff
@@ -0,0 +1,198 @@
+// bdlb_pcgrandomgenerator.h                                                         -*-C++-*-
+#ifndef INCLUDED_BDLB_PCGRANDOMGENERATOR
+#define INCLUDED_BDLB_PCGRANDOMGENERATOR
+
+#include <bsls_ident.h>
+BSLS_IDENT("$Id: $")
+
+//@PURPOSE: Provide a class to generate random numbers using the PCG algorithm.
+//
+//@CLASSES:
+//  bdlb::PCGRandomGenerator: random number generator
+//
+//@SEE_ALSO: bdlb_random
+//
+//@DESCRIPTION: This component provides a single class,
+// 'bdlb::PCGRandomGenerator', that is used to generate random numbers
+// employing the PCG algorithm, a high-performance, high-quality RNG.  The PCG
+// technique employs the concepts of permutation functions on tuples and a base
+// linear congruential generator.
+//
+//
+///Usage
+///-----
+// This section illustrates intended use of this component.
+//
+///Example 1: Generating random 32-bit numbers
+///- - - - - - - - - - - - - - - - - - - - - -
+// The PCGRandomGenerator constructor takes two 64-bit constants (the initial
+// state, and the rng sequence selector.  Instances of PCGRandomGenerator with
+// different sequence selectors will never have random sequences that coincide.
+// In order to obtain nondeterministic output for each instance, on a Unix
+// system random bytes can be obtained from /dev/urandom.  The component 
+// bdlb_randomdevice provides such functionality.  Also illustrated below is a 
+// a backup mechanism for obtaining random bytes.
+//
+//  uint64_t state;
```

Comment:
```suggestion
//..
//  uint64_t state;
```

Reply:
We'll bring the wording to more BDE style later.  For now, just couple formatting nits.